app/src/interfaces/presentation-links/presentation-links.vue

```diff
@@ -15,65 +13,96 @@ type Link = {
 	url?: string;
 };
 
+type ParsedLink = Omit<Link, 'url'> & {
+	to?: string;
+	href?: string;
+};
+
 type Props = {
 	links: Link[];
 	collection: string;
-	primaryKey: string;
+	primaryKey?: PrimaryKey;
 };
 
 const props = withDefaults(defineProps<Props>(), {
 	links: () => [],
 });
 
+const api = useApi();
 const values = inject('values', ref<Record<string, any>>({}));
-
-const { collection, primaryKey } = toRefs(props);
-
-const query = computed(() => {
-	const fields = new Set();
-
-	props.links.forEach((link) => {
-		getFieldsFromTemplate(link.url ?? '').forEach((field) => fields.add(field));
-	});
-
-	return {
-		fields: Array.from(fields),
-	} as Query;
-});
-
-const { item } = useItem(collection, primaryKey, query);
-const { fields } = useCollection(collection);
-
-const fullItem = computed(() => {
-	const itemValue = item.value ?? {};
-
-	for (const field of fields.value) {
-		if (
-			field.meta?.special?.some((special) => RELATIONAL_TYPES.includes(special as (typeof RELATIONAL_TYPES)[number]))
-		) {
-			continue;
-		}
-
-		itemValue[field.field] = values.value[field.field];
+const resolvedRelationalValues = ref<Record<string, any>>({});
+
+/**
+ * Get all deduplicated relational fields from the link-templates.
+ * For example:
+ * [ "related.field", "languages.code" ]
+ */
+const relatedFieldsFromTemplates = computed(
+	() =>
+		props.links
+			?.flatMap((link) => {
+				if (!link.url) return [];
+				return (
+					getFieldsFromTemplate(link.url)
+						// filter out any duplicates for this link
+						.filter((value, index, array) => array.indexOf(value) === index)
+						// filter out non-relations, since they should be included in the values already
+						.filter((value) => value.includes('.'))
+				);
+			})
+			// filter out any duplicates between all links
+			.filter((value, index, array) => array.indexOf(value) === index),
+);
+
+watch(relatedFieldsFromTemplates, async () => {
+	// No need to fetch if there are no fields or we're creating a new item
+	if (relatedFieldsFromTemplates.value.length === 0 || props.primaryKey === '+' || !props.primaryKey) return;
+
+	try {
+		const response = await api.get(`${getEndpoint(props.collection)}/${props.primaryKey}`, {
+			params: {
+				fields: relatedFieldsFromTemplates.value,
+			},
+		});
+
+		// Pick only non-arrays because we cant render those types of relations
+		// For example a M2M relation will return an array
+		resolvedRelationalValues.value = pickBy(response.data.data, (value) => !Array.isArray(value));
+	} catch (err) {
+		// eslint-disable-next-line no-console
+		console.warn('Presentation-Link: Fetching related fields failed');
```

Comment:
Might be better to use `unexpectedError` instead of the `console.warn` so the user would see the error in the Data Studio